test/infrastructure/docker/e2e/docker_machine_remediation_test.go

```diff
@@ -0,0 +1,192 @@
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/test/framework"
+	infrav1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1alpha3"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Docker Recover", func() {
+	var (
+		replicas     = 3
+		namespace    = "default"
+		clusterGen   = newClusterGenerator("recover")
+		mgmtClient   ctrlclient.Client
+		cluster      *clusterv1.Cluster
+		controlPlane *controlplanev1.KubeadmControlPlane
+	)
+	SetDefaultEventuallyTimeout(10 * time.Minute)
+	SetDefaultEventuallyPollingInterval(10 * time.Second)
+
+	BeforeEach(func() {
+		// Ensure multi-controlplane workload cluster is up and running
+		var (
+			infraCluster *infrav1.DockerCluster
+			template     *infrav1.DockerMachineTemplate
+			err          error
+		)
+		cluster, infraCluster, controlPlane, template = clusterGen.GenerateCluster(namespace, int32(replicas))
+
+		// Set up the client to the management cluster
+		mgmtClient, err = mgmt.GetClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		// Set up the cluster object
+		createClusterInput := framework.CreateClusterInput{
+			Creator:      mgmtClient,
+			Cluster:      cluster,
+			InfraCluster: infraCluster,
+		}
+		framework.CreateCluster(ctx, createClusterInput)
+
+		// Set up the KubeadmControlPlane
+		createKubeadmControlPlaneInput := framework.CreateKubeadmControlPlaneInput{
+			Creator:         mgmtClient,
+			ControlPlane:    controlPlane,
+			MachineTemplate: template,
+		}
+		framework.CreateKubeadmControlPlane(ctx, createKubeadmControlPlaneInput)
+
+		// Wait for the cluster to provision.
+		assertClusterProvisionsInput := framework.WaitForClusterToProvisionInput{
+			Getter:  mgmtClient,
+			Cluster: cluster,
+		}
+		framework.WaitForClusterToProvision(ctx, assertClusterProvisionsInput)
+
+		// Wait for at least one control plane node to be ready
+		waitForOneKubeadmControlPlaneMachineToExistInput := framework.WaitForOneKubeadmControlPlaneMachineToExistInput{
+			Lister:       mgmtClient,
+			Cluster:      cluster,
+			ControlPlane: controlPlane,
+		}
+		framework.WaitForOneKubeadmControlPlaneMachineToExist(ctx, waitForOneKubeadmControlPlaneMachineToExistInput, "5m")
+
+		// Insatll a networking solution on the workload cluster
+		workloadClient, err := mgmt.GetWorkloadClient(ctx, cluster.Namespace, cluster.Name)
+		Expect(err).ToNot(HaveOccurred())
+		applyYAMLURLInput := framework.ApplyYAMLURLInput{
+			Client:        workloadClient,
+			HTTPGetter:    http.DefaultClient,
+			NetworkingURL: "https://docs.projectcalico.org/manifests/calico.yaml",
+			Scheme:        mgmt.Scheme,
+		}
+		framework.ApplyYAMLURL(ctx, applyYAMLURLInput)
```

Comment:
I've noticed we're installing calico in every test, we could switch to kindnet. @fabriziopandini found out that to be much faster to get nodes to `Ready`

Reply:
We should definitely explore this if it is faster for the new e2e implementation.